### ISSUE TITLE
Build and publish muzzle check plugin separately

### ISSUE THREAD

**Issue by iNikem** [open-telemetry/opentelemetry-java-instrumentation#1225]
When trying to run muzzle in parallel in GHA, building projects `buildSrc` takes 2/3 of each task execution. This is total waste of time. Should build it once, publish and consume externally.

**Comment by iNikem** [comment]
Damn :( muzzle plugin, living inside `buildSrc`, reaches to `javaagent-bootstrap` and `javaagent-tooling` to grab their runtime classpaths :( This makes extracting plugin almost impossible :(

**Comment by pavolloffay** [comment]
Is there any way how it could be done? I am happy to help with this.

**Comment by iNikem** [comment]
I don't know one right now.

**Comment by pavolloffay** [comment]
@iNikem Is this the problem?

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/4624584979c87dde9e90c8fc2275201601d4d798/buildSrc/src/main/groovy/MuzzlePlugin.groovy#L62-L63

I would like to use the muzzle plugin in a custom build with added instrumentations. The muzzle would be very helpful there.

**Comment by iNikem** [comment]
Yes, this is it.

**Comment by pavolloffay** [comment]
Commenting from https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/1408

The muzzle task could expose a property to configure classpath with `javaagent-tooling` and `javaagent-bootstrap` artifacts. E.g something similar to what ByteBuddy plugin does
https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/master/gradle/instrumentation.gradle#L36

```groovy
  byteBuddy {
    transformation {
      tasks = ['compileJava', 'compileScala', 'compileKotlin']
      plugin = 'io.opentelemetry.javaagent.tooling.muzzle.collector.MuzzleCodeGenerationPlugin'
      classPath = project(':javaagent-tooling').configurations.instrumentationMuzzle + configurations.runtimeClasspath + sourceSets.main.output
    }
  }
```

**Comment by mateuszrzeszutek** [comment]
A bit of clarification: this issue concerns the muzzle check plugin (`MuzzlePlugin` currently), not the muzzle compile one - there's a separate issue for the muzzle compile plugin extraction.

**Comment by iNikem** [comment]
Done
